src/main/java/me/botsko/prism/commands/WandCommand.java

```diff
@@ -17,262 +22,253 @@
-					if (InventoryUtils.handItemToPlayer(inv, new ItemStack(item_material, 1))) {
-						wand.setItemWasGiven(true);
-					}
-					else {
-						call.getPlayer().sendMessage(
-								Prism.messenger.playerError("Can't fit the wand item into your inventory."));
-					}
-				}
-				InventoryUtils.updateInventory(call.getPlayer());
-			}
-
-			// Let's build the QueryParameters for it if it's a Query wand.
-			if (wand instanceof QueryWandBase) {
-				if (!((QueryWandBase) wand).setParameters(call.getPlayer(), call.getArgs(), (isInspect ? 1 : 2))) { // This
-																													// returns
-																													// if
-																													// it
-																													// was
-																													// successful
-					call.getPlayer().sendMessage(Prism.messenger.playerError("Notice: Only some parameters used.."));
-				}
-			}
-
-			// Store
-			Prism.playersWithActiveTools.put(call.getPlayer().getName(), wand);
-		}
-		else {
-			if (oldwand != null) {
-				oldwand.disable(call.getPlayer());
-			}
-		}
-	}
-
-	@Override
-	public List<String> handleComplete(CallInfo call) {
-		return null;
-	}
+    /**
+     *
+     */
+    private final Prism plugin;
+
+    /**
+     * @param plugin
+     * @return
+     */
+    public WandCommand(Prism plugin) {
+        this.plugin = plugin;
+    }
+
+    /**
+     * Handle the command
+     */
+    @Override
+    public void handle(CallInfo call) {
+        String type = "i";
+        final boolean isInspect = call.getArg(0).equalsIgnoreCase("inspect") || call.getArg(0).equalsIgnoreCase("i");
+        if (!isInspect) {
+            if (call.getArgs().length < 2) {
+                call.getPlayer().sendMessage(
+                        Prism.messenger.playerError("You need to specify a wand type. Use '/prism ?' for help."));
+                return;
+            }
+            type = call.getArg(1);
+        }
+
+        Wand oldwand = null;
+        if (Prism.playersWithActiveTools.containsKey(call.getPlayer().getName())) {
+            // Pull the wand in use
+            oldwand = Prism.playersWithActiveTools.get(call.getPlayer().getName());
+        }
+
+        // Always remove the old one
+        Prism.playersWithActiveTools.remove(call.getPlayer().getName());
+
+        // Determine default mode
+        String mode = plugin.getConfig().getString("prism.wands.default-mode");
+
+        // Check if the player has a personal override
+        if (plugin.getConfig().getBoolean("prism.wands.allow-user-override")) {
+            final String personalMode = Settings.getSetting("wand.mode", call.getPlayer());
+            if (personalMode != null) {
+                mode = personalMode;
+            }
+        }
+
+        // Determine which item we're using.
+        String toolKey = null;
+        if (mode.equals("item")) {
+            toolKey = plugin.getConfig().getString("prism.wands.default-item-mode-id");
+        } else if (mode.equals("block")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)